MoltenVK/MoltenVK/GPUObjects/MVKInstance.mm

```diff
@@ -345,6 +346,42 @@
 	}
 }
 
+// If the VK_EXT_layer_settings extension is enabled, initialize the local
+// MVKConfiguration from the global version built from environment variables.
+void MVKInstance::initMVKConfig(const VkInstanceCreateInfo* pCreateInfo) {
+
+	if ( !_enabledExtensions.vk_EXT_layer_settings.enabled ) { return; }
+
+	_mvkConfig = mvkConfig();
+
+	VkLayerSettingsCreateInfoEXT* pLSCreateInfo = nil;
+	for (const auto* next = (VkBaseInStructure*)pCreateInfo->pNext; next; next = next->pNext) {
+		switch (next->sType) {
+			case VK_STRUCTURE_TYPE_LAYER_SETTINGS_CREATE_INFO_EXT:
+				pLSCreateInfo = (VkLayerSettingsCreateInfoEXT*)next;
+				break;
+			default:
+				break;
+		}
+	}
+
+	if ( !pLSCreateInfo ) { return; }
+
+	for (uint32_t lsIdx = 0; lsIdx < pLSCreateInfo->settingCount; lsIdx++) {
+		const auto* pSetting = &pLSCreateInfo->pSettings[lsIdx];
+
+#define STR(name) #name
+#define MVK_CONFIG_MEMBER(member, mbrType, name) \
+		if(mvkStringsAreEqual(pSetting->pLayerName, getDriverLayer()->getName()) &&  \
+		   mvkStringsAreEqual(pSetting->pSettingName, STR(MVK_CONFIG_##name))) {  \
+			_mvkConfig.member = *(mbrType*)(pSetting->pValues);  \
+			continue;  \
+		}
```

Comment:
> I don't think that VK_EXT_layer_settings imposes that requirement, though.

It doesn't you are correct this is an issue.

Reply:
Good catch. That is the reason for adding the overloaded `mvkSetConfig()` that takes a `std::string`. It's called just below the code above (and in the global config too). For `MVKInstance` it's held in the `std::string _autoGPUCaptureOutputFilepath` member variable.

It's kludgy because it presupposes we have only one string config value, and if we get many more, we'll have to be more elegant about it...but I figured those will be rare...or might never happen.

Reply:
> It's kludgy because it presupposes we have only one string config value, and if we get many more

Ideally an assert is added to prevent an unpleasant debugging session.

Reply:
> Ideally an assert is added to prevent an unpleasant debugging session.

What are you thinking we should assert?

Reply:
Count the number of VK_LAYER_SETTING_TYPE_STRING_EXT used. If it exceeds more than 1 assert.

Reply:
> Count the number of VK_LAYER_SETTING_TYPE_STRING_EXT used. If it exceeds more than 1 assert.

Ah. I decided to go full out. I added support for future multiple strings, and added assertions to ensure they are handled properly.

Reply:
@cdavis5e Can you have another look, please, and approve if it all looks good now.